### PR TITLE
Fixes for three more pe corner cases

### PR DESCRIPTION
**Pull request by hillu**
See #441

### CODE REVIEW

File: libyara/modules/pe.c [hunk 1]
```diff
@@ -1052,7 +1052,7 @@ int pe_valid_dll_name(
     }
   }
 
-  return (l > 0 && l < n);
+  return (l > 0 && l <= n);
```
Comment (plusvic) [inline libyara/modules/pe.c]: I think this introduces a potential buffer overrun in line 1117, where yr_strdup is called with dll_name as its argument. If dll_name is at the end of the file and without a null termination, pe_valid_dll_name will return true and yr_strdup can be invoked with a non-null-terminated string.

Comment (hillu) [inline libyara/modules/pe.c]: You're correct, of course. Turns out that the use of a `strdup` style function is wrong here. But then, we didn't know about the string-at-EOF corner case before...

File: libyara/modules/pe.c [hunk 2]
```diff
@@ -438,22 +431,9 @@ int64_t pe_rva_to_offset(
...
-    section_raw_size = pe->data_size;
-  }
-
-  // Many sections, have a raw (on disk) size smaller than their in-memory size.
-  // Check for rva's that map to this sparse space, and therefore have no valid
-  // associated file offset.
-
-  if ((rva - section_rva) >= section_raw_size)
-    return -1;
+  // Apparently, the loader maps data that resides between sections in
```
Comment (plusvic) [inline libyara/modules/pe.c]: The returned offset is going to be equal to the passed RVA for any RVA not contained within a section. What if the value passed to pe_rva_to_offset is an offset within the PE overlay? The value is going to be accepted as a valid RVA and is going to be returned as the corresponding offset. But I think the PE overlay is not mapped into memory by the loader? I don't have a Windows machine around to verify it. Can you confirm if that's the intended behavior?

**Comment by hillu** [comment]
Yeah, .I  going to do some experiments.

**Pull request closed without merging**